src/main/java/com/unstoppabledomains/resolution/naming/service/uns/L2Resolver.java

```diff
@@ -0,0 +1,57 @@
+package com.unstoppabledomains.resolution.naming.service.uns;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import com.unstoppabledomains.exceptions.ns.NSExceptionCode;
+import com.unstoppabledomains.exceptions.ns.NSExceptionParams;
+import com.unstoppabledomains.exceptions.ns.NamingServiceException;
+
+/**
+ * 
+ */
+public class L2Resolver {
+  private ExecutorService executor = Executors.newFixedThreadPool(2);
+
+  public <T> List<T> resolveOnBothLayers(ResolutionMethods<T> methods) throws NamingServiceException{
+    Future<T> l1result = executor.submit(methods.getL1Func());
+    Future<T> l2result = executor.submit(methods.getL2Func());
+    ArrayList<T> results = new ArrayList<>();
+    results.add(processFutureResult(l1result));
+    results.add(processFutureResult(l2result));
+    return results;
+  }
+  
+  public <T> T resolve(ResolutionMethods<T> methods) throws NamingServiceException{
+    Future<T> l1result = executor.submit(methods.getL1Func());
+    Future<T> l2result = executor.submit(methods.getL2Func());
+
+    try {
+      return processFutureResult(l2result);
+    } catch (NamingServiceException e) {
+      if (e.getCode() != NSExceptionCode.UnregisteredDomain && e.getCode() != NSExceptionCode.UnspecifiedResolver) {
```

Comment:
@DancingAxolotl If domain doesn't have specified resolver this means it exists on L2

Reply:
If the resolver isn't specified, we cannot get any data from the domain. It's not really a network error, so I assumed that we should try to get data from L1 in this case.

Should we throw this as an error instead?

Reply:
@DancingAxolotl Yes, we should throw an error instead. If resolver isn't specified or if we can't find domain record, but L2 domain has owner (i.e. is minted on L2) this means that we should use L2 response and ignore L1 response. We should fallback to L1 response only if domain is not minted on L2 (which means domain doesn't have an owner).

Reply:
@DancingAxolotl Please make sure that all methods of UnsInternal class will throw correct error in case if domain doesn't have an owner. It's even better to add tests for such cases to all UnsInternal methods.

Reply:
I've updated the L2Resolver logic and double-checked `UNSInternal` and `ResolutionTest` to make sure that correct errors are thrown.